### PR TITLE
Add pre merge checks - fix jobs syntax in workflow

### PR DESCRIPTION
**Pull request by nizamutdinovb**

### CODE REVIEW

**Comment by nizamutdinovb** [comment]
@rjhuijsman I have made typo in not specifying `jobs` in the workflow.
This new PR is going to fix it.
Could you please approve this PR as well in order to merge it to main and made the workflow working?

**Comment by rjhuijsman** [comment]
> @rjhuijsman Also I noticed that `PRIVATE_REPO_ACCESS_AS_REBOT_TOKEN` secret is not present in secrets. image
> 
> Currently the secrets are configured to be used only with private repos. But `pyprotoc-plugin` is a public one. So now we have two options:
> 
> 1. to configure the secrets to be used by all repos
> 2. to change workflow to not use `PRIVATE_REPO_ACCESS_AS_REBOT_TOKEN` since there are no submodules in it
> 
> @rjhuijsman what do you think of it?

That's a great question! Let's go with option 2: unlike our other repos `pyprotoc-plugin` doesn't currently use submodules, so we should be able to clone the repository with the default `$GITHUB_TOKEN` (you can probably omit the entire `token` configuration entirely).

**Review by rjhuijsman — CHANGES_REQUESTED** [review]

File: .github/workflows/pre-merge-checks.yml [hunk 1]
```diff
@@ -7,14 +7,15 @@ on:
...
+jobs:
+  run-tests:
+    name: Run pre-merge checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout reboot-dev/respect
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.PRIVATE_REPO_ACCESS_AS_REBOT_TOKEN }}
+          submodules: recursive
```
[the remark this answers is not in the document]
Comment (nizamutdinovb) [reply .github/workflows/pre-merge-checks.yml]: Done. Now it's fine. Review re-requested.

**Comment by rjhuijsman** [comment]
@nizamutdinovb the reason this pipeline is now failing is that we require Python version `>=3.9` for our tests to work. The default on GitHub action runners is 3.8.

How easy/hard would it be to make our GitHub Action Runner use Python 3.9?

**Comment by nizamutdinovb** [comment]
> @nizamutdinovb the reason this pipeline is now failing is that we require Python version `>=3.9` for our tests to work. The default on GitHub action runners is 3.8.
> 
> How easy/hard would it be to make our GitHub Action Runner use Python 3.9?

@rjhuijsman I have installed python3.9 to the Runner during the Build before running the tests but it failed again:
https://github.com/reboot-dev/pyprotoc-plugin/runs/6197505246?check_suite_focus=true
and
https://github.com/reboot-dev/pyprotoc-plugin/runs/6197571899?check_suite_focus=true

**Comment by rjhuijsman** [comment]
Replaced by https://github.com/reboot-dev/pyprotoc-plugin/pull/25

**Pull request closed without merging**